tests/platform_tests/api/test_component.py

```diff
@@ -70,11 +68,10 @@ def compare_value_with_platform_facts(self, key, value, component_idx):
             if expected_components:
                 expected_value = expected_components[component_idx].get(key)
 
-        if not expected_value:
-            logger.warning("Unable to get expected value for '{}' from platform.json file for component {}".format(key, component_idx))
-            return
+        pytest_assert(expected_value is not None,
+                      "Unable to get expected value for '{}' from platform.json file for component {}".format(key, component_idx))
 
-        self.expect(value == expected_value,
+        pytest_assert(value == expected_value,
```

Comment:
You should use a nested `self.expect()` here, as this gets called on a number of devices in the list, this way we can gather all the error messages. Please refer ot my SFP changes in https://github.com/Azure/sonic-mgmt/pull/2268, not the chassis changes.